api/app/utils/cffdrs.py

```diff
@@ -0,0 +1,109 @@
+""" This module contains functions for computing fire weather metrics.
+"""
+import logging
+from rpy2.robjects.packages import importr
+cffdrs = importr('cffdrs')
+
+logger = logging.getLogger(__name__)
+
+#   From cffdrs R package comments:
+#   FUELTYPE: The Fire Behaviour Prediction FuelType
+#        ISI: Initial Spread Index
+#        BUI: Buildup Index
+#        FMC: Foliar Moisture Content
+#        SFC: Surface Fuel Consumption (kg/m^2)
+#         PC: Percent Conifer (%)
+#        PDF: Percent Dead Balsam Fir (%)
+#         CC: Constant (crown closure)
+#        CBH: Crown to base height(m)
+
+# Returns:
+#   ROS: Rate of spread (m/min)
+#
+
+# Computable: SFC, FMC
+# To store in DB: PC, PDF, CC, CBH (attached to fuel type, red book)
+PARAMS_ERROR_MESSAGE = "One or more params passed to R call is None."
+
+DEFAULT_METRICS = {"PC": 1, "PDF": 1, "CC": 1, "CBH": 1}
+
+# TODO: Once PC, PDF, CC, CDH are confirmed, replace these defaults
+# and potentially store then in the DB as columns in FuelType
+FUEL_TYPE_LOOKUP = {"C1": DEFAULT_METRICS,
+                    "C2": DEFAULT_METRICS,
+                    "C3": DEFAULT_METRICS,
+                    "C4": DEFAULT_METRICS,
+                    "C5": DEFAULT_METRICS,
+                    "C6": DEFAULT_METRICS,
+                    "C7": DEFAULT_METRICS,
+                    "D1": DEFAULT_METRICS,
+                    "D2": DEFAULT_METRICS,
+                    "M1": DEFAULT_METRICS,
+                    "M2": DEFAULT_METRICS,
+                    "M3": DEFAULT_METRICS,
+                    "M4": DEFAULT_METRICS,
+                    "O1A": DEFAULT_METRICS,
+                    "O1B": DEFAULT_METRICS,
+                    "S1": DEFAULT_METRICS,
+                    "S2": DEFAULT_METRICS,
+                    "S3": DEFAULT_METRICS
+                    }
+
+
+def rate_of_spread(fuel_type: str, isi: float, bui: float, fmc: float, sfc: float):
+    """ Computes ROS by delegating to cffdrs R package """
+    if fuel_type is None or isi is None or bui is None or sfc is None:
+        logger.error(
+            "%s fuel_type: %s. isi: %s, bui: %s, sfc: %s",
+            PARAMS_ERROR_MESSAGE, fuel_type, isi, bui, sfc)
+        return None
```

Comment:
How about raising an exception instead of returning None? I like this level of testing that's being added, the R code is pretty loosy goosy.